src/test/java/uk/gov/hmcts/ccd/endpoint/exceptions/ApiExceptionTest.java

```diff
@@ -0,0 +1,122 @@
+package uk.gov.hmcts.ccd.endpoint.exceptions;
+
+import org.junit.Test;
+import uk.gov.hmcts.ccd.domain.model.common.CatalogueResponse;
+import uk.gov.hmcts.ccd.domain.model.common.CatalogueResponseElement;
+
+import java.util.Arrays;
+import java.util.List;
+
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.not;
+import static org.junit.Assert.*;
+
+public class ApiExceptionTest {
+
+    @Test
+    public void shouldExtractMessageFromCatalogueResponse() {
+
+        // ARRANGE
+        final CatalogueResponse testCatalogueResponse =
```

Comment:
there is little value in having `final` in test classes. It makes the code very heavy to read in my opinion. I would remove it. Would also remove it in general. It make the code unclean for very little benefit